plugins/lombok/lombok-compiler-plugin/src/org/jetbrains/kotlin/lombok/LombokSyntheticJavaPartsProvider.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.lombok
+
+import org.jetbrains.kotlin.descriptors.ClassConstructorDescriptor
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.descriptors.SimpleFunctionDescriptor
+import org.jetbrains.kotlin.load.java.lazy.descriptors.LazyJavaClassDescriptor
+import org.jetbrains.kotlin.load.java.structure.impl.JavaClassImpl
+import org.jetbrains.kotlin.lombok.config.LombokConfig
+import org.jetbrains.kotlin.lombok.processor.*
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.resolve.jvm.SyntheticJavaPartsProvider
+import java.util.*
+
+/**
+ * Provides synthetic parts to java classes (from current compilation unit), which will be generated by lombok AnnotationProcessor
+ * So kotlin can reference lombok members.
+ */
+class LombokSyntheticJavaPartsProvider(config: LombokConfig) : SyntheticJavaPartsProvider {
+
+    private val processors = listOf(
+        GetterProcessor(config),
+        SetterProcessor(config),
+        WithProcessor(),
+        NoArgsConstructorProcessor(),
+        AllArgsConstructorProcessor(),
+        RequiredArgsConstructorProcessor()
+    )
+
+    /**
+     * kotlin resolve references in two calls - first it gets names, then actual member descriptor
+     * but for us it is much easier to run full generation for class once
+     * hence we cache results and reuse it
+     */
+    private val partsCache: MutableMap<ClassDescriptor, SyntheticParts> = WeakHashMap()
```

Comment:
I'm not sure I understand,
LombokSyntheticJavaPartsProvider will be created for each module if https://github.com/JetBrains/kotlin/pull/4262/files#diff-df4021efbb6de59a6b576250bb90dc4799b9847c0f71788abb564f66917b2decR120 is called for each module 
and if it is created for each module it won't remove the need of this map. Basically map is for convenience caching. And it is 'weak' because I'm not use how long my instance will live inside DI context (probably not needed)